### PR TITLE
[NOJIRA] Update Deps and Add (Jest + other dotfiles)

### PR DESCRIPTION
**Pull request by jamesz96**
- Update circle CI
- add npmrc
- add nvmrc
- add jest
- add new tests

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,15 +1,14 @@
 {
   "compilerOptions": {
-    "outDir": "./lib",
-    "rootDir": "./src",
-    "target": "es5",
-    "declaration": true,
-    "lib": ["dom", "es2017"],
```
Comment (jamesz96) [inline tsconfig.json:7]: Hmm do I need to keep "dom" here. 
🤔

Comment (marhalpert) [reply tsconfig.json:7]: Is this imported in to any www portal?

Comment (marhalpert) [reply tsconfig.json:7]: I think you need to keep it old browser friendly
Screen Shot 2021-12-15 at 10 41 42 am

Comment (jamesz96) [reply tsconfig.json:7]: library is used by both customer and admin portals. 
I'll re-add it.

Comment (marhalpert) [reply tsconfig.json:7]: I wonder if when we import it into the customer portal we compile it again with webpack 🤔

Comment (jamesz96) [reply tsconfig.json:7]: I'm not too familiar with webpack. How should I go about checking this?

File: tsconfig.json [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+    "target": "es2020",
```
Comment (marhalpert) [inline tsconfig.json:3]: Should you targer es5? like the other libs?

**Review by marhalpert — APPROVED** [review]

File: tsconfig.json (hunk 2 above)
Comment (jamesz96) [reply tsconfig.json]: updated

**Pull request merged**